### ISSUE TITLE
[Question] How To Transparent ?

### ISSUE THREAD

**Issue by MisterTeeXD** [OpenShot/openshot-qt#2264]
Hello, i need to know how can i make 2 animated images transparent?

**Comment by peanutbutterandcrackers** [comment]
I wonder if this isn't quite possible.

There is `apng` that does support transparency. But your animated images are in `gif` format, probably. And even that isn't supported by OpenShot, yet.

However, if you really wanted to take the trouble of doing it, you could:
1. Take note of the FPS of the animated image.
2. Break it into single frames (I think [GIMP](http://gimp.org/) can handle that)
3. Using GIMP, erase the background of every single frame (but the subject you want to have moving), probably using the [foreground select](https://www.youtube.com/watch?v=uhRGix-x5Mg&t=368s) tool (in 2.10.x series, the 2.8.x series' fore-ground selection tool isn't as good) and then export each frame as a png file (with transparency).
4. Import those into OpenShot as an image sequence and set their frame rate to be the same as the original file's frame rate.

Or, if your animated image has an object moving in a relatively solid coloured background, I suppose you could import them as an image sequence and then apply the [chroma key](https://www.youtube.com/watch?v=2sushecqMs4&t=340s) effect on them to key out the background. But I am still not quite sure.

Hope that helps!

**Comment by MisterTeeXD** [comment]
I still think Openshot needs a transparent background, it would help so much

**Comment by peanutbutterandcrackers** [comment]
I don't fully understand. Do you want the tracks below the image to be shown? Or do you want to export a video that contains transparency? I don't know much about them. But I did find [this discussion on stack exchange](https://superuser.com/questions/746232/why-are-there-no-video-formats-that-support-transparency).

Perhaps a description of what you want to achieve (along with a screenshot or some other graphics) will help me understand this better?

**Comment by MisterTeeXD** [comment]
Whoops, let me try again, the preview in Openshot has a black background which does not look like transparency, i have key-framed two images in Openshot that represents a transition for an open source software called OBS Studio, upon importing that animation, it does not look transparent when it has been activated and OBS Studio has a feature that allows you to select the Stinger Transition type as choice for transition and allows you to use a animated video for a transition method but does not support effects, what i suggest is, make the background of the preview transparent by default so that way, any keyframed Images has a transparent background by default if rendered with the correct Format which can then be used in OBS Studio for Transition effects, from what i heard is that the "webm" format supports transparency.

**Comment by DylanC** [comment]
@MisterTeeXD - I don't really understand this question. Do you have a picture to explain or a link to something descriptive in relation to this?

**Comment by DreamZuLu** [comment]
@DylanC 

I have the same problem, I would like to have a Transparent background option in Openshot 
- Ill explain it like this:

"When you export an empty video with on content at all, only audio perhaps. Then you get a video with a black background - What we want is for the background to be standard Transparent or have an option to chose color or transparent background" :)

If you have any contacts, it seems like an easy fix ^^

**Comment by ModMailGrAdient** [comment]
warning

**Comment by ferdnyc** [comment]
OpenShot does not currently support a transparent background on its output video, and to my knowledge has no plans to at this time. Most formats have _no_ support for this. A few do (Chrome can play WebM video with an alpha channel, it's true), but even then preparing the video is usually a matter of performing background-removal (AKA green-screening) on a non-transparent video file. Actual transparency in the final output is better suited to SVG animation or web canvas rendering than full-motion video; OpenShot is simply not the right tool for that job.

Using a **solid** color _other than_ black as the background for a video in OpenShot is easy, though. Just make a PNG file the size of your output profile, fill it with the color you want as your background, and drop it on Track 1 of the timeline.

**Comment by cepacolmax** [comment]
Ok, but hear me out - I have an animated APNG that I use as an episode title, with a little whooshing sound that goes along with it. 
I used APNG because it lays over the action  for which it is the title - the animation is on the top layer, and you can see the action behind it. 

Getting the audio track lined up with it was kind of fiddly, so I'd like to package the APNG and the audio track together so it's easy to use in future episodes.
 
It seems natural to render the animation and audio into an MP4 that can be easily imported into each new project, laid over the current episode's action, but if that kills the transparency that's no bueno. 

So, if that's not the solution, what is? How does one group together a collection of clips in a way that makes them easy to manipulate as a unit and re-use from one project to the next, while preserving the transparency?

**Comment by ferdnyc** [comment]
@cepacolmax 

That's a fair question. Currently there isn't a completely effortless way to do that in OpenShot. It's a task generally better suited to an application like After Effects, rather than a NLE for full-motion video like OpenShot.

However, OpenShot does include a form of support for rendered animations, in the form of the Animated Titles editor (which creates animations from rendered Blender scenes), so you could borrow from there.

If you explode your APNG into individual PNG images (with transparency), frame-by-frame, then you could import the series as an image sequence. That'd allow you to use it exactly the way you describe. OpenShot **does** support transparency for still images, and what is an animation really but a series of still frames shown in sequence? (This is exactly how the rendered Blender animations from the Animated Titles editor are imported, in fact.)

You wouldn't be able to keep the audio _with_ the images directly, but that's the only limitation, and can probably be worked around. I'd think that once you have them lined up, as long as you trim the audio and the image sequence so they start at exactly the same point, then it should be easy enough to align them together using the timeline snapping feature. (Or, in fact, you could use the Align options in the context menu, which are added once you've selected more than one clip.) If you have to add some silence to the beginning of the audio file, or some blank frames at the start of the image sequence, so that they line up again... then do that, so they're as easy as possible to align.

So, yeah, that's what I'd recommend currently. An image sequence of transparent animation frames, and a corresponding audio file trimmed so their start points line up.

Using a video format mostly wouldn't work anyway, or it would be subject to severe limitations. Like, for instance, H.264 has no support for transparency. It's not part of the codec. Nothing we could do to make it work. I don't think H.265 supports it either. So while it might theoretically be possible to create a transparent MP4 (I'm not sure whether the container itself would interfere, TBH), it'd require using an alternate codec. Transparency support for full-motion video codecs is _extremely_ limited even today.

**Comment by zavcmlopez** [comment]
You say "OpenShot does support transparency for still images" How can I use that using the Openshot cloud API?

Thanks a lot

Transparent still images are good  for me

**Comment by JacksonRG** [comment]
@zavcmlopez Hi, not to keep you waiting. I'm making a project in the desktop app with a transparency effect, compressing it, and uploading it to the api.

That way, I'll be able to see how the effect is formatted in the api.

Anything you'd like to do in the API should be able to be done in the desktop app, and uploaded. If you can't wait for my answer, I'd encourage you to try that.

**Comment by JacksonRG** [comment]
@zavcmlopez 
Here's the json I used in the api to set a transparent image.

the `Points -> co -> x` value is the frame that you want alpha value to be whatever the `y` is. (Alpha means opacity)
the interpolation option controls how the alpha value will change between the keyframes. (0 is a bezier curve, 1 is a linear change, 2 is contant value).

```json
    "alpha": {
      "Points": [
        {
          "co": {
            "X": 1,
            "Y": 1
          },
          "handle_left": {
            "X": 0.5,
            "Y": 1
          },
          "handle_right": {
            "X": 0.42,
            "Y": 0.0
          },
          "handle_type": 1,
          "interpolation": 0
        },
        {
          "co": {
            "X": 241,
            "Y": 0.0
          },
          "interpolation": 0,
          "handle_left": {
            "Y": 1.0,
            "X": 1.0
          }
        }
      ]
    },
```

If you want the whole image to be transparent, without changing over time, I'd just set a point `x:0`, `y: <your value less than 1`,
and interpolation 2 makes it constant. (though unless you have a different alpha value later in the clip that shouldn't matter)

There's more info on this topic here:
https://cloud.openshot.org/doc/animation.html#key-frames
